src/Microsoft.AspNetCore.Razor/CodeGenerators/CSharpTagHelperCodeRenderer.cs

```diff
@@ -450,112 +449,90 @@ protected virtual string GenerateUniqueId()
             }
         }
 
-        private void RenderUnboundAttribute(string attributeName, Chunk attributeValueChunk)
+        private void RenderUnboundAttribute(TagHelperAttributeTracker attribute)
         {
             // Render children to provide IntelliSense at design time. No need for the execution context logic, it's
             // a runtime feature.
             if (_designTimeMode)
             {
-                if (attributeValueChunk != null)
+                if (attribute.Value != null)
```

Comment:
Check still needs to exist here because pre-allocation logic is for runtime. At design time we don't do any of the work to determine if something needs to be pre-allocated for runtime (doesn't have any value).

Will add a != null check after the `_designTimeMode` statement